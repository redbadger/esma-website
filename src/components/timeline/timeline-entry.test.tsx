import React from "react";
import renderer from "react-test-renderer";
import TimelineEntry from "./timeline-entry";
import { useStaticQuery } from "gatsby";

describe("TimelineEntry", () => {
  it("renders correctly", () => {
    useStaticQuery.mockImplementation(() => {
      return testData;
    });
    const tree = renderer
      .create(
        <TimelineEntry
          imageName="test-image.jpg"
          title="test"
          description="test"
          icon={Icon}
          destination="/"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const Icon = () => {
  return <img />;
};

const testData = {
  images: {
    edges: [
      {
        node: {
          childImageSharp: {
            fluid: {
              base64: "data:image/jpeg;base64,/9j/2wBDABALDA4M...",
              sizes: "(max-width: 800px) 100vw, 800px",
              src: "/static/test/path1/test-image.jpg",
              srcSet: "/static/test/path1/test-image.jpg 200w",
              aspectRatio: 1.5,
              originalName: "test-image.jpg",
            },
          },
        },
      },
    ],
  },
};
