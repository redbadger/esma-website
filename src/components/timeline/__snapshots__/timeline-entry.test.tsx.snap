// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineEntry renders correctly 1`] = `
<a>
  <div
    className="css-82p6gm-arrowHoverCss-timelineEntryStyles"
  >
    <div
      className="timeline-image-wrapper gatsby-image-wrapper"
      style={
        Object {
          "maxHeight": null,
          "maxWidth": null,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "66.66666666666667%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        aria-hidden={true}
        src="data:image/jpeg;base64,/9j/2wBDABALDA4M..."
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 800px) 100vw, 800px"
          srcSet="/static/test/path1/test-image.jpg 200w"
        />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 800px) 100vw, 800px"
          src="/static/test/path1/test-image.jpg"
          srcSet="/static/test/path1/test-image.jpg 200w"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/test/path1/test-image.jpg 200w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/test/path1/test-image.jpg 200w\\" src=\\"/static/test/path1/test-image.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
    <h2>
      test
    </h2>
    <img />
    <div
      className="separator"
    />
    <p>
      test
    </p>
    <div
      className="read-more"
    >
      Read more
      <span
        className="arrow css-84es3n-arrowCss"
      />
    </div>
  </div>
</a>
`;
